app-backend/common/src/main/scala/cache/HeapBackedMemcachedClient.scala

```diff
@@ -0,0 +1,62 @@
+package com.azavea.rf.common.cache
+
+import com.github.blemale.scaffeine.{ Cache => ScaffeineCache, Scaffeine }
+import net.spy.memcached._
+
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.util.matching._
+
+
+/**
+  * Wraps a MemcachedClient to provide a local cache which stores Future results in case
+  *  the call has recently been made and could otherwise result in a race condition.
+  *
+  *  @note Each instance of this wrapper will have its own instance of a Caffeine
+  *         cache (a guava-inspired java library).
+  */
+class HeapBackedMemcachedClient[CachedType](
+  client: MemcachedClient,
+  options: HeapBackedMemcachedClient.Options = HeapBackedMemcachedClient.Options()) {
+
+  /** The caffeine cache (on heap) which prevents race conditions */
+  val onHeapCache: ScaffeineCache[String, Future[CachedType]] =
+    Scaffeine()
+      .recordStats()
+      .expireAfterAccess(options.ttl)
+      .maximumSize(options.maxSize)
+      .build[String, Future[CachedType]]()
+
+  /**
+    * The primary means of interaction with this class: pass as key and
+    *  the costly option caching prevents duplication of.
+    */
+  def caching(cacheKey: String)(expensiveOperation: String => Future[CachedType])(implicit ec: ExecutionContext): Future[CachedType] = {
```

Comment:
What is the purpose of getting the cache key back to the `expensiveOperation`? The caller must have passed it, so it must already be there. It doesn't seem like any of the current uses of the cache benefit from it either.
If there is no clear purpose for it I would suggest `expensiveOperation: => Future[CachedType]`.